tests/conftest.py

```diff
@@ -947,53 +981,29 @@ def _make_libp2p_connection(
         f"./{agent_key.address}_cert.txt",
     )
     _process_cert(agent_key, cert_request, path_prefix=data_dir)
-    if not build_directory:
-        build_directory = os.getcwd()
+
+    build_directory = build_directory or os.getcwd()
     config = {"ledger_id": node_key.identifier}
-    if relay and delegate:
-        configuration = ConnectionConfig(
-            node_key_file=node_key_file,
-            local_uri="{}:{}".format(host, port),
-            public_uri="{}:{}".format(host, port),
-            entry_peers=entry_peers,
-            log_file=log_file,
-            delegate_uri="{}:{}".format(delegate_host, delegate_port),
-            peer_registration_delay=peer_registration_delay,
-            connection_id=P2PLibp2pConnection.connection_id,
-            build_directory=build_directory,
-            cert_requests=[cert_request],
-            **config,  # type: ignore
-        )
-    elif relay and not delegate:
-        configuration = ConnectionConfig(
-            node_key_file=node_key_file,
-            local_uri="{}:{}".format(host, port),
-            public_uri="{}:{}".format(host, port),
-            entry_peers=entry_peers,
-            log_file=log_file,
-            peer_registration_delay=peer_registration_delay,
-            connection_id=P2PLibp2pConnection.connection_id,
-            build_directory=build_directory,
-            cert_requests=[cert_request],
-            **config,  # type: ignore
-        )
-    else:
-        configuration = ConnectionConfig(
-            node_key_file=node_key_file,
-            local_uri="{}:{}".format(host, port),
-            entry_peers=entry_peers,
-            log_file=log_file,
-            peer_registration_delay=peer_registration_delay,
-            connection_id=P2PLibp2pConnection.connection_id,
-            build_directory=build_directory,
-            cert_requests=[cert_request],
-            **config,  # type: ignore
-        )
+    port = port or next(default_ports)
+    configuration = ConnectionConfig(
+        local_uri=f"{host}:{port}",
+        entry_peers=entry_peers,
+        log_file=log_file,
+        peer_registration_delay=peer_registration_delay,
+        connection_id=P2PLibp2pConnection.connection_id,
+        build_directory=build_directory,
+        cert_requests=[cert_request],
+        **config,  # type: ignore
+    )
 
+    if relay:
+        configuration.config["public_uri"] = f"{host}:{port}"
+    if delegate:
+        delegate_port = delegate_port or next(default_ports)
+        configuration.config["delegate_uri"] = f"{delegate_host}:{delegate_port}"
     if mailbox:
+        mailbox_port = mailbox_port or next(default_ports)
```

Comment:
apart from being clearer, we now also no longer increase the `default_port` counter in case we're not actually using the delegate or mailbox port